common/src/warp_runner/manager/commands/other_commands.rs

```diff
@@ -0,0 +1,107 @@
+use std::{
+    fs::File,
+    io::{Read, Seek, Write},
+    path::{Path, PathBuf},
+    thread,
+};
+use walkdir::WalkDir;
+use zip::{result::ZipError, write::FileOptions};
+
+use derive_more::Display;
+use futures::channel::oneshot;
+use warp::{error, logging::tracing::log};
+#[derive(Display)]
+pub enum OtherCmd {
+    #[display(fmt = "CompressFolder {{ src: {src:?}, dest: {dest:?} }} ")]
+    CompressFolder {
+        src: PathBuf,
+        dest: PathBuf,
+        rsp: oneshot::Sender<Result<(), error::Error>>,
+    },
+}
+
+pub async fn handle_other_cmd(cmd: OtherCmd) {
+    match cmd {
+        OtherCmd::CompressFolder { src, dest, rsp } => {
+            let r = compress_folder(src, dest).await;
+            let _ = rsp.send(r);
+        }
+    }
+}
+
+async fn compress_folder(src: PathBuf, dest: PathBuf) -> Result<(), error::Error> {
+    let (tx, rx) = oneshot::channel::<Result<(), ZipError>>();
+    // I know that warp_runner is basically single threaded but still...put the blocking operation in a separate thread and await it
+    thread::spawn(move || {
```

Comment:
you could technically get away with using `tokio::spawn_blocking` since it would use tokio own threadpool to spawn a thread for blocking operations but would allow it to yield back to the executor so there technically wouldnt be any need for the oneshot channel here.